src/ebfm/LOOP_EBM_insolation.py

```diff
@@ -71,7 +69,7 @@ def main(grid, time2, OUT):
         np.sin(lat_rad) * np.sin(d_rad) + np.cos(lat_rad) * np.cos(d_rad) * np.cos(h_rad)
     )  # SOURCE: Iqbal (1983)
 
-    if grid["input_type"] is GridInputType.MATLAB:  # only use shading routine in uncoupled runs on a regular grid
+    if grid["has_shading"]:
         yl = grid["Ly"]
         xl = grid["Lx"]
```

Comment:
Instead of checking for the `GridInputType` I would generally check whether shading is supported.